File: OrderServer/OrderServer.API/OrderServer.Shared/Dispatcher/Queries/QueryDispatcher.cs
using Microsoft.Extensions.DependencyInjection;
using OrderServer.Shared.Queries;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrderServer.Shared.Dispatcher.Queries
{
    public class QueryDispatcher : IQueryDispatcher
    {
        private readonly IServiceProvider _serviceProvider;
        public QueryDispatcher(
            IServiceProvider serviceProvider)
        {
            _serviceProvider= serviceProvider;
        }
        public Task<TResult> DispatchAsync<TResult>(IQuery<TResult> query) where TResult : class
        {
            using var scope = _serviceProvider.CreateScope();
            // make type
            var handlerType = typeof(IQueryHandler<,>).MakeGenericType(query.GetType(), typeof(TResult));
            var handler = scope.ServiceProvider.GetService(handlerType);

            // get method
            var handleAsyncMethod = handlerType.GetMethod(nameof(IQueryHandler<IQuery<TResult>,TResult>.QueryAsync))!;

            return (Task<TResult>)handleAsyncMethod.Invoke(handler, new object[] { query })!;
        }
    }
}

File: MovieServer/MovieServer.API/packages/MovieServer.Shared/src/Dispatcher/Queries/QueryDispatcher.ts
import { Injectable, InversifyContainer } from "../../IoC";
import { GetSymbol } from "../../IoC/Utils";
import { IQuery, IQueryHandler } from "../../Queries";
import { Container } from "inversify";
import IQueryDispatcher from "./IQueryDispatcher";

@Injectable
export default class QueryDispatcher implements IQueryDispatcher {
    private readonly _container: Container;
    constructor(
        @InversifyContainer container: Container
    ) {
        this._container = container;

        this.ExecuteAsync = this.ExecuteAsync.bind(this);
    }
    ExecuteAsync<TResult>(query: IQuery<TResult>): Promise<TResult | null> {
        console.log(`Dispatcher: IQueryHandler<${query.constructor.name}>`);
        const symbol = GetSymbol(`IQueryHandler<${query.constructor.name}>`);
        const handler = this._container.get<IQueryHandler<IQuery<TResult>, TResult>>(symbol as symbol);
        return handler.HandleAsync(query);
    }

}